lib/replace-in-file.js

```diff
@@ -60,33 +61,49 @@ module.exports = function replaceInFile(config, cb) {
   }
 
   //Initialize helper vars
-  var totalFiles = config.files.length;
+  var totalFiles = 0;
   var processedFiles = 0;
   var changedFiles = [];
 
   //No callback given? Perform sync operation
   if (!cb) {
     config.files.forEach(function(file) {
-      if (replaceSync(file, config.replace, config.with)) {
-        changedFiles.push(file);
-      }
+      //Expand globs
+      glob.sync(file, {nodir: true}).forEach(function(file) {
+        if (replaceSync(file, config.replace, config.with)) {
+          changedFiles.push(file);
+        }
+      });
     });
     return changedFiles;
   }
 
   //Otherwise, perform async operation
   config.files.forEach(function(file) {
-    replace(file, config.replace, config.with, function(error, hasChanged) {
-      if (error) {
-        return cb(error);
-      }
-      if (hasChanged) {
-        changedFiles.push(file);
+    //Expand globs
+    glob(file, {nodir: true}, function(err, files) {
+      if (err) {
+        return cb(err);
       }
-      processedFiles++;
-      if (processedFiles === totalFiles) {
-        cb(null, changedFiles);
+      //Glob doesn't error if no files match, so we do:
+      if (files.length === 0) {
+        return cb(new Error('No files match the pattern: ' + file));
       }
+      files.forEach(function(file) {
+        totalFiles++;
```

Comment:
>  am I missing something?

@adambuczynski `files` in this context is an array of files expanded from a single glob. I am using nested `forEach` blocks here. The outer block starts at: https://github.com/adambuczynski/replace-in-file/pull/8/files#diff-2fa7042c579966cbf571668ebb660e6fR82.